### ISSUE TITLE
Replace hardcoded context roots from app-cfg.js

### ISSUE THREAD

**Issue by zwets** [flowable/flowable-engine#1022]
When deploying the Flowable UI Apps under a different context root from the default `/flowable-*`, it is possible to make everything work using changes in property files (e.g. `flowable.common.app.idm-url` is configurable), except for one party pooper: the file `./src/main/resources/static/scripts/app-cfg.js` has hardcoded context root strings:

```
FLOWABLE.CONFIG = {
        'onPremise' : true,
        'contextRoot' : '/flowable-idm',
        'webContextRoot' : '/flowable-idm',
    'datesLocalization' : false
};
```

If these static strings weren't there, then the app wars could be transparently deployed under a different context root. Currently, putting an app on a different context root requires modifying the war file.

**Comment by PascalSchumacher** [comment]
Indeed, it would be nice to have this changed/fixed.

**Comment by tijsrademakers** [comment]
If there are ideas on how to do this in a different way please add it here to the PR.

**Comment by ruzkant** [comment]
How about making it a JAX-RS resource and replace variable? Or use spring mvc templates, but that's an additional dependency probably.

**Comment by paulstapleton** [comment]
I've come across this problem before as well and fixed it with a little JavaScript in` app-cfg.js`. It's a bit of a hacky solution where the context is parsed from the URL stored in the browser, but it has been working for me when using an alternative context name in tomcat.

There may well be problems with it that I am not aware of, but I'll leave the code here and you can decide if its something that is usable or not.

```
var FLOWABLE = FLOWABLE || {};

FLOWABLE.CONFIG = function($) {

    var config = $("script[src*='app-cfg.js']");
    var context = config[0].src.replace(location.origin, "").replace("/scripts/app-cfg.js", "").split("?", 1)[0];

    return {
           'onPremise' : true,
           'contextRoot' : context,
           'webContextRoot' : context,
           'datesLocalization' : false
     };

}(jQuery);
```

**Comment by PascalSchumacher** [comment]
No idea if this is a good solution (I'm not a front-end devleoper), what about using:

`var baseURL = window.location.protocol + '//' + window.location.host;`

source https://stackoverflow.com/questions/35116446/angular-relative-paths-in-http-requests

**Comment by PascalSchumacher** [comment]
I guess this can be closed now that #1100 has been merged?

**Comment by tijsrademakers** [comment]
Thanks for the ping, this is implemented with the PR indeed.
